src/scales/scale.time.js

```diff
@@ -176,6 +176,19 @@ module.exports = function(Chart) {
 			Chart.Scale.prototype.initialize.call(this);
 		},
 
+        /**
+         * Allow x-axis to be accessed via x or t attribute
+         */
+		getRightValue: function(rawValue) {
+			if (typeof(rawValue) === 'object') {
+				if ((rawValue instanceof Date) || rawValue.isValid) {
```

Comment:
Good call on moving the date handling from core to time. I've done that. I didn't make the other changes. partially because `helpers.isObject` seems bugged. https://github.com/chartjs/Chart.js/issues/4538